### ISSUE TITLE
Allow users to upload images via a drag and drop

### ISSUE THREAD

**Issue by rolandgeider** [wger-project/react#353]
As an improvement, we could allow users to upload images via a drop zone

![image](https://user-images.githubusercontent.com/2280729/181925714-7ac9af7a-b231-44db-956f-56fb0ab8b5c0.png)

**Comment by Ruchit1111** [comment]
Hello Rolandgeider,
 I am interested to contribute in this.

**Comment by rolandgeider** [comment]
Hi @Ruchit1111 ! Sure, the issue is yours. You can reach the form from training > Add new exercise

You can use the testserver as a backend, just set this in your .env.development (note that the database is reset every day):

```
REACT_APP_API_SERVER=https://wger.rge.uber.space
REACT_APP_API_KEY=778a360dbf79f602f9ef72027067b3117f778111
```

**Comment by Saurabhbangar0** [comment]
I want to work on this issue. if it's okay,assign it to me.

**Comment by rolandgeider** [comment]
@Ruchit1111 are you working on this?

**Comment by Ruchit1111** [comment]
> @Ruchit1111 are you working on this?

Yes I am working on it I have run the project using the key you have provided

**Comment by Saurabhbangar0** [comment]
yes

**Comment by faizan7800** [comment]
Hi @rolandgeider! I noticed this issue has been open for a while with no linked PR. I'm a frontend developer and would love to work on this. Is it still available? I was thinking of using react-dropzone library to implement the drop zone. Happy to discuss the approach first if needed!

**Comment by rolandgeider** [comment]
hey @faizan7800 ! Yes this is still available. I'm not sure right now if we share any components between the submission page and the regular edit page. But it would be useful to be able to drag images there as well.

**Comment by faizan7800** [comment]
Hi @rolandgeider, thanks for assigning me! I'm already exploring the codebase to understand how the image upload currently works on both pages. Will update you once I have a clear picture of the approach

**Comment by faizan7800** [comment]
Hi @rolandgeider, after exploring the codebase I found that ImageDropZone component already exists in src/components/Exercises/forms/ImageDropZone.tsx and is already integrated into ImageFormModal which is used on both the submission page (Step5Images) and the edit page (ExerciseDetailEdit). It supports drag & drop, click to select, image preview, and file type validation. Could you clarify what's still missing or what improvements you'd like to see? Happy to work on whatever is needed!

**Comment by rolandgeider** [comment]
oh you are right, sorry! I guess we missed linking the PR, closing here.

(btw if you want to contribute other things, give me a couple of days till I plan out the next steps)

**Comment by faizan7800** [comment]
Thanks for the quick response! No worries at all. I'll keep an eye out for the next steps, looking forward to contributing more!

**Comment by rolandgeider** [comment]
@faizan7800 FYI the next things planned for 2.7 are in [this project](https://github.com/orgs/wger-project/projects/8/views/1), the main hone being the [health data sync](https://github.com/wger-project/wger/issues/2417)

**Comment by faizan7800** [comment]
Thanks for sharing the roadmap!  and sorry for the late response from my
side as I was on a trip with my family. I'll take a look at the project
board and see where I can contribute. Really appreciate you keeping me in
the loop!
